src/BulletHell/PowerManager.java

```diff
@@ -0,0 +1,85 @@
+package BulletHell;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import BulletHell.Powerups.*;
+import edu.macalester.graphics.CanvasWindow;
+import edu.macalester.graphics.events.Key;
+
+/**
+ * A class responsible for adding the power boxes to the canvas, and binding the powers with their respective keys.
+ */
+public class PowerManager {
+    public static final int GAP = 50;
+
+    private List<Powerups> allPowerups = new ArrayList<>();
+    private List<Key> activateKeys = new ArrayList<>();
```

Comment:
Keep a `Map<Key,Power>` instead of two separate lists. Among other things, that would simplify `activatePowerups` quite a bit.
```suggestion
    private List<Powerups> allPowerups = new ArrayList<>();
    private List<Key> activateKeys = new ArrayList<>();
```
```suggestion
    private List<Powerups> allPowerups = new ArrayList<>();
    private List<Key> activateKeys = new ArrayList<>();
```